platform/linux-generic/odp_packet_io.c

```diff
@@ -932,34 +932,86 @@ static inline int pktin_recv_buf(pktio_entry_t *entry, int pktin_index,
 	return num_rx;
 }
 
+static inline int packet_vector_send(odp_queue_t queue, odp_event_t event)
+{
+	odp_packet_vector_t pktv = odp_packet_vector_from_event(event);
+	odp_packet_t *pkt_tbl;
+	int num, sent;
+
+	num = odp_packet_vector_tbl(pktv, &pkt_tbl);
+	ODP_ASSERT(num > 0);
+	sent = odp_pktout_send(queue_fn->get_pktout(queue), pkt_tbl, num);
+
+	/* Return success if any packets were sent. Free the possible remaining
+	   packets in the vector. */
+	if (odp_unlikely(sent <= 0))
+		return -1;
+	else if (odp_unlikely(sent != num))
+		odp_packet_free_multi(&pkt_tbl[sent], num - sent);
```

Comment:
In this case, aren't we dropping packets without any errors or counters visible to the application?

Reply:
When using odp_queue_enq() to send packets there may be intermediate steps (e.g. internal reorder queue) before the packets are actually transmitted to the wire, so trusting to the function return value is alone is difficult. The correct error counters should be incremented on the pktio driver level. I'll go through the drivers and add counters where necessary.